stdlib/oauth2/src/main/ballerina/src/oauth2/inbound_oauth2_provider.bal

```diff
@@ -208,6 +230,8 @@ public type IntrospectionServerConfig record {|
     http:ClientConfiguration clientConfig = {};
 |};
 
+// Deprecated: This record was used for OAuth2 caching and with the new cache API v2.0.0 this record no longer used
```

Comment:
We have a deprecated annotation in Ballerina. Shall we use that
https://ballerina.io/learn/by-example/deprecation.html

Reply:
Yes. Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/24490/commits/a1deb5f79cf4df524fd4942a01718eac94ae8e48